packages/vite-plugin/src/plugin-utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { Stylable, StylableMeta } from '@stylable/core';
+import type { StylableExports } from '@stylable/core/dist/index-internal';
+import { processUrlDependencies } from '@stylable/build-tools';
+import fs from 'fs';
+import { basename, extname, isAbsolute, join } from 'path';
+import { createHash } from 'crypto';
+import mime from 'mime';
+import type { Plugin } from 'vite';
+import type { StylableVitePluginOptions } from './index';
+
+type ObjectHook<T, O = {}> = T | ({ handler: T; order?: 'pre' | 'post' | null } & O);
+type GetThisParameters<T extends (this: any, ...args: any) => any> = T extends (
+    this: infer P,
+    ...args: any
+) => any
+    ? P
+    : never;
+
+type VitePluginTransfrom = GetThisParameters<
+    Plugin['transform'] extends ObjectHook<infer T, {}> ? NonNullable<T> : false
+>;
+
+export function generateStylableModuleCode(
```

Comment:
notice that on master this was refactored to a common place between integrations